core/camel-util/src/main/java/org/apache/camel/util/ClassLoadingAwareObjectInputStream.java

```diff
@@ -58,6 +60,14 @@ protected Class<?> resolveClass(ObjectStreamClass classDesc) throws IOException,
         return load(classDesc.getName(), cl, inLoader);
     }
 
+    class NoopDynamicInvocationHandler implements InvocationHandler {
+
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+            return null;
+        }
+    }
```

Comment:
What is the reason to use a specific NoopInvocationHandler?
Why not using org.apache.camel.component.bean.CamelInvocationHandler?

Reply:
In this case we just a need a dummy Invocationhandler, we are only interested in the Proxy class, not the proxy itself, anyway, I did refactor this to avoid potential NPEs